Automattic/jetpack
# Create JS Config package

## The problem we are trying to solve

We need a way for the consumer of the Jetpack packages (for example the stand-alone plugins) to declare some information to be available to our all packages.

For example, we'd like that the `api` package to be aware of what's the consumer that is performing the API requests. For example, when the `ConnectButton` makes a request to the `v4/register` endpoint, we'd like to know if this request is coming from the Jetpack or Backup plugin.

This package solves this by creating an interface to register variables that our packages can read.

This will be done by adding an `externals` entry in the webpack configuration file. This will act like a module and make an object available to all our packages.

The goal of this Config package is to serve as a proxy to this configuration object, in a way we can throw explanatory errors and hold the documentation on how to use it.

## Other approaches tried

See https://github.com/Automattic/jetpack/pull/21563

#### Changes proposed in this Pull Request:

* Creates the JS Config package

#### Jetpack product discussion


See https://github.com/Automattic/jetpack/pull/21563

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Look at the code, tests, and discuss the approach
* Look at https://github.com/Automattic/jetpack/pull/21593 to see how this is going to be used
